From 10191d783bee09033bafef81e5611b69c8ca4c16 Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Thu, 22 Nov 2018 18:22:31 +0900
Subject: [PATCH 363/909] media: rcar-vin: Add CPG reset and release when using
 digital pin

commit 3351cd0ee31cb1b5fd70233b9318f35e37d63a51 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

According to the R-Car Gen3 Hardware Manual Errata for Rev 1.00 of
Jun 2, 2018, The procedure of CPG reset and release was added when
using digital pin. This patch adds it. In addition, review camera
start flow and CPG reset and release.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/rcar-vin/rcar-v4l2.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/media/platform/rcar-vin/rcar-v4l2.c b/drivers/media/platform/rcar-vin/rcar-v4l2.c
index 598d1439ae9f..0c23ab84aa7a 100644
--- a/drivers/media/platform/rcar-vin/rcar-v4l2.c
+++ b/drivers/media/platform/rcar-vin/rcar-v4l2.c
@@ -914,6 +914,10 @@ static int rvin_power_on(struct rvin_dev *vin)
 	ret = v4l2_subdev_call(sd, core, s_power, 1);
 	if (ret < 0 && ret != -ENOIOCTLCMD && ret != -ENODEV)
 		return ret;
+
+	reset_control_assert(vin->rstc);
+	reset_control_deassert(vin->rstc);
+
 	return 0;
 }
 
-- 
2.17.1

